cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/TopicReplicaDistributionGoal.java

```diff
@@ -37,9 +37,7 @@
 
 
 /**
- * Class for achieving the following soft goal:
- * <p>
- * SOFT GOAL#2: Balance collocations of replicas of the same topic.
+ * SOFT GOAL: Balance collocations of replicas of the same topic.
```

Comment:
Can you explain how this goal avoid move non-immigrant replica to other broker? 
I thought there would be some logic in [numReplicasToMove](https://github.com/linkedin/cruise-control/blob/1ca1898e24434cc9a842a263f7814bf6e8218f45/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ReplicaDistributionTarget.java#L184)...

Reply:
The current implementation of this goal only balances the replicas on dead brokers if there are dead brokers (see `TopicReplicaDistributionGoal#brokersToBalance(ClusterModel)`) -- i.e. it guarantees that no non-offline replica will be moved in case there are dead brokers. Going forward, we should improve this goal to consider immigrant replicas. (relevant ticket: https://github.com/linkedin/cruise-control/issues/624)